### PR TITLE
Fix DASession and NSMutableArray leaks on macOS

### PR DESCRIPTION
**Pull request by zvin**
Change-type: patch

### CODE REVIEW

File: src/darwin/list.mm [hunk 1]
```diff
@@ -131,6 +131,7 @@ DeviceDescriptor CreateDeviceDescriptorFromDiskDescription(std::string diskBsdNa
       CFRelease(diskDescription);
       CFRelease(disk);
     }
+    [dl dealloc];
```
Comment (zrzka) [inline src/darwin/list.mm]: You're not supposed to call `dealloc` - never. The only place where you can do this is inside `dealloc` method where you must call `[super dealloc]` as the last thing. Should be `release`.

**Comment by zrzka** [comment]
Missing `[dl release]` [here](https://github.com/balena-io-modules/drivelist/blob/929fd6110d1f3eb9bf7bdcc3d27b6fd02f5c6a93/src/darwin/list.mm#L108).

File: src/darwin/list.mm (hunk 1 above)
Comment (zvin) [reply src/darwin/list.mm]: amended

**Comment by zvin** [comment]
>Missing [dl release] here.
@zrzka moved the init after `DASessionCreate` instead.

**Pull request merged**